client/src/components/FormBuilderModal/FormBuilderModal.js

```diff
@@ -1,104 +1,71 @@
-import React, { Component } from 'react';
-import i18n from 'i18n';
-import { Modal, ModalHeader } from 'reactstrap';
+import React, { useState } from 'react';
 import FormBuilderLoader from 'containers/FormBuilderLoader/FormBuilderLoader';
 import castStringToElement from 'lib/castStringToElement';
 import classnames from 'classnames';
 import PropTypes from 'prop-types';
+import Modal from 'components/Modal/Modal';
+import pick from 'lodash/pick';
 
 const noop = () => null;
 
-class FormBuilderModal extends Component {
-  constructor(props) {
-    super(props);
-
-    this.handleSubmit = this.handleSubmit.bind(this);
-    this.handleHide = this.handleHide.bind(this);
-    this.clearResponse = this.clearResponse.bind(this);
-    this.handleLoadingError = this.handleLoadingError.bind(this);
-  }
-
-  /**
-   * Defines the form part of the Modal
-   *
-   * @returns {Component}
-   */
-  getForm() {
-    const { FormBuilderLoaderComponent } = this.props;
-    if (!this.props.schemaUrl) {
-      return null;
-    }
-    return (
-      <FormBuilderLoaderComponent
-        fieldHolder={{ className: classnames('modal-body', this.props.bodyClassName) }}
-        actionHolder={{ className: 'modal-footer' }}
-        autoFocus={this.props.autoFocus}
-        schemaUrl={this.props.schemaUrl}
-        onSubmit={this.handleSubmit}
-        onAction={this.props.onAction}
-        onLoadingError={this.handleLoadingError}
-        identifier={this.props.identifier}
-      />
-    );
-  }
-
-  /**
-   * Generates the response part of the Modal
-   *
-   * @returns {Component}
-   */
-  getResponse() {
-    if (!this.state || !this.state.response) {
-      return null;
-    }
-
-    let className = '';
-
-    if (this.state.error) {
-      className = this.props.responseClassBad;
-    } else {
-      className = this.props.responseClassGood;
-    }
-
-    return (
-      <div className={className}>
-        { castStringToElement('span', { html: this.state.response }) }
-      </div>
-    );
-  }
-
-  /**
-   * Removes the response from the state
-   */
-  clearResponse() {
-    this.setState({
-      response: null,
-    });
-  }
-
-  handleLoadingError(schema) {
-    const providesOnLoadingError = this.props.onLoadingError !== noop;
-    if (this.props.showErrorMessage || !providesOnLoadingError) {
-      const error = schema.errors && schema.errors[0];
-      this.setState({
-        response: error.value,
-        error: true,
-      });
+/**
+ * React component for displaying a Form in a Modal using a form schema URL
+ */
+function FormBuilderModal(props) {
+  const {
+    responseClassBad,
+    responseClassGood,
+    onLoadingError,
+    showErrorMessage,
+    onClosed,
+    onSubmit,
+    schemaUrl,
+    bodyClassName,
+    FormBuilderLoaderComponent,
+    children
+  } = props;
+
+  const [response, setResponse] = useState(null);
+  const [error, setError] = useState(null);
+
+  // Register a success message
+  const setSuccess = (message) => {
```

Comment:
```suggestion
  const handleSuccess = (message) => {
```
Prefixing with `set` is confusing as it makes it sound like we're updating a state hook

Update the other 2 functions below

Reply:
`setSuccess` and `setFailure` are updating state hooks. They are essentially hooks.

I've moved all the `useState` logic to its own function. This as the added benefit of making a somewhat long component a bit smaller and manageable.

Reply:
Oh right it's in a custom hook

The use of a custom is honestly just confusing, it's adding complexity where it doesn't need to be added, it's not making things any simpler

For instance `setSuccess()` is called once. However there are 5 matches in the file if I search for `setSuccess`

Remove the hook and just put the state directly in the component

Reply:
`clearResponse` was part a method on the class component.

I created the other method for symmetry's sake. But I can see the point that it's overkilled in this context.

I ditched setSuccess, setFailure and clearResponse hide and manually update the state each time.